packages/entity/src/__tests__/EntityQueryContext-test.ts

```diff
@@ -0,0 +1,102 @@
+import invariant from 'invariant';
+
+import { EntityQueryContext } from '../EntityQueryContext';
+import ViewerContext from '../ViewerContext';
+import { createUnitTestEntityCompanionProvider } from '../utils/testing/createUnitTestEntityCompanionProvider';
+
+describe(EntityQueryContext, () => {
+  describe('callbacks', () => {
+    it('calls all callbacks, and calls invalidation first', async () => {
+      const companionProvider = createUnitTestEntityCompanionProvider();
+      const viewerContext = new ViewerContext(companionProvider);
+
+      let preCommitFirstCallCount = 0;
+      let preCommitSecondCallCount = 0;
+      let postCommitCallCount = 0;
+      let postCommitInvalidationCallCount = 0;
+
+      const preCommitFirstCallback = async (): Promise<void> => {
+        preCommitFirstCallCount++;
+      };
+      const preCommitSecondCallback = async (): Promise<void> => {
+        preCommitSecondCallCount++;
+      };
```

Comment:
Could do `const preCommitFirstCallback = jest.fn(async() => {})` and then assert invocation counts with `preCommitFirstCallback.mock.calls.length` or `expect(preCommitFirstCallback).toHaveBeenCalledTimes(n)`